records/generator_test.go

```diff
@@ -250,87 +251,63 @@ func TestInsertState(t *testing.T) {
 
-	// ensure we are only collecting running tasks
-	_, ok := rg.SRVs["_poseidon._tcp.marathon.mesos."]
-	if ok {
-		t.Error("should not find this not-running task - SRV record")
-	}
-
-	rrs, ok := rg.As["liquor-store.marathon.mesos."]
-	if !ok {
-		t.Error("should find this running task - A record")
-	}
-	if got, want := rrs, []string{"1.2.3.11", "1.2.3.12"}; !reflect.DeepEqual(got, want) {
-		t.Errorf("should return the slave ips 1.2.3.11, 1.2.3.12 for the task, but got %v", rrs)
-	}
-
-	rrs, ok = rg.As["_container.liquor-store.marathon.mesos."]
-	if !ok {
-		t.Error("should find the container ip")
-	}
-	if got, want := rrs, []string{"10.3.0.1", "10.3.0.2"}; !reflect.DeepEqual(got, want) {
-		t.Errorf("should return the container IPs 10.3.0.1, 10.3.0.2 for the task, but got %v", rrs)
-	}
-
-	_, ok = rg.As["poseidon.marathon.mesos."]
-	if ok {
-		t.Error("should not find this not-running task - A record")
-	}
-
-	_, ok = rg.As["_container.poseidon.marathon.mesos."]
-	if ok {
-		t.Error("should not find a container IP")
-	}
-
-	_, ok = rg.As["master.mesos."]
-	if !ok {
-		t.Error("should find a running master - A record")
-	}
-
-	_, ok = rg.As["master0.mesos."]
-	if !ok {
-		t.Error("should find a running master0 - A record")
-	}
-
-	_, ok = rg.As["leader.mesos."]
-	if !ok {
-		t.Error("should find a leading master - A record")
-	}
-
-	_, ok = rg.SRVs["_leader._tcp.mesos."]
-	if !ok {
-		t.Error("should find a leading master - SRV record")
-	}
-
-	// test for 10 SRV names
-	if got, want := len(rg.SRVs), 10; got != want {
-		t.Errorf("not enough SRVs, got %d, expected %d", got, want)
-	}
-
-	// test for 5 A names
-	if got, want := len(rg.As), 16; got != want {
-		t.Errorf("not enough As, got %d, expected %d", got, want)
-	}
-
-	// ensure we translate the framework name as well
-	_, ok = rg.As["some-box.chronoswithaspaceandmixe.mesos."]
-	if !ok {
-		t.Error("should find this task w/a space in the framework name - A record")
-	}
-
-	// ensure we find this SRV
-	rrs = rg.SRVs["_liquor-store._tcp.marathon.mesos."]
-	// ensure there are 3 RRDATA answers for this SRV name
-	if len(rrs) != 3 {
-		t.Error("not enough SRV records")
-	}
+	type expectation struct {
+		domain string
+		num    uint
+		answer []string
+		reason string
+	}
+	expectedARecords := []expectation{
+		{domain: "liquor-store.marathon.mesos.", num: 2, answer: []string{"1.2.3.11", "1.2.3.12"}},
+		{domain: "_container.liquor-store.marathon.mesos.", num: 2, answer: []string{"10.3.0.1", "10.3.0.2"}},
+		{domain: "_container.reviewbot.marathon.mesos.", num: 1, answer: []string{"10.5.0.7"}},
+		{domain: "poseidon.marathon.mesos.", reason: "task was not running"},
+		{domain: "_container.poseidon.marathon.mesos.", reason: "task was not running"},
+		{domain: "master.mesos.", num: 1},
+		{domain: "master0.mesos.", num: 1},
+		{domain: "leader.mesos.", num: 1},
+		{domain: "some-box.chronoswithaspaceandmixe.mesos.", num: 1}, // ensure we translate the framework name as well
```

Comment:
In table driven tests, it isn't strictly necessary to explicitly name the struct keys. It actually hinders readability IMO.